uxarray/utils/helpers.py

```diff
@@ -673,3 +678,114 @@ def close_face_nodes(Mesh2_face_nodes, nMesh2_face, nMaxMesh2_face_nodes):
     np.put(closed.ravel(), first_fv_idx_1d, first_node_value)
 
     return closed
+
+
+def point_within_GCR(pt, gcr_cart):
+    """Check if a point is on a given Great Circle Arc. The anti-meridian case
+    is already considered.
+
+    Parameters
+    ----------
+    pt : np.ndarray of float
+        Cartesian coordinates of the point
+    gcr_cart : np.ndarray of shape (2, 3), (np.float or gmpy2.mpfr)
+        Cartesian coordinates of the GCR
+
+    Returns
+    -------
+    bool
+        True if the point is on the GCR (Lies between the two endpoints), False otherwise
+    """
+    # Convert the cartesian coordinates to lonlat coordinates, the node_xyz_to_lonlat_rad is already overloaded
+    # with gmpy2 data type
+    pt_lonlat = node_xyz_to_lonlat_rad(pt)
+    GCRv0_lonlat = node_xyz_to_lonlat_rad(gcr_cart[0])
+    GCRv1_lonlat = node_xyz_to_lonlat_rad(gcr_cart[1])
+
+    # First if the input GCR is exactly 180 degree, we throw an exception, since this GCR can have multiple planes
+    angle = _angle_of_2_vectors(gcr_cart[0], gcr_cart[1])
+    if np.allclose(angle, np.pi, rtol=0, atol=ERROR_TOLERANCE):
+        raise ValueError(
+            "The input GCR is exactly 180 degree, this GCR can have multiple planes. Consider breaking the GCR "
+            "into two GCRs")
+
+    if not np.allclose(np.dot(np.cross(gcr_cart[0], gcr_cart[1]), pt),
+                       0,
+                       rtol=0,
+                       atol=ERROR_TOLERANCE):
+        return False
```

Comment:
Can you explain the purpose of this block?

Reply:
So this step is to make sure that the point is on the same plane defined by the two endpoints of the arcs:

The math concept can easily understood here: [reference](https://users.math.msu.edu/users/gnagy/teaching/11-fall/mth234/l06-234.pdf)
<img width="666" alt="image" src="https://github.com/UXARRAY/uxarray/assets/76922794/8a899b19-a6c1-4384-a461-086a5285b176">

P is our intersection point and P0 is our sphere's original point.

At last, we know the it's non-realistic to attain an absolute 0 in the program, we allow a tolerance.

Reply:
Thank you for clarifying!